zenoh_router/datadog_checks/zenoh_router/check.py

```diff
@@ -0,0 +1,112 @@
+            else:
+                res[(name, '')] = val
+        return res
+
+    def __map_to_ddtags(self, m):
+        return [k + ':' + v for k, v in m.items()]
+
+    def __process_stats(self, stats, tags, rid):
+        cached_values = self.__cache.get(rid)
+
+        timestamp = time.time()
+        elapsed_seconds = timestamp - self.__cache[rid]['timestamp']
+        self.__cache[rid]['timestamp'] = timestamp
+
+        val = self.__parse_stats(stats)
+        for k, v in val.items():
+            v = float(v)
+            if cached_values is not None and k in cached_values['metrics']:
+                d = (v - cached_values['metrics'][k]) / elapsed_seconds
+                if d < 0:
+                    d = 0
+                name, space = k
+                tags['space'] = space
+                self.gauge(name, d, tags=self.__map_to_ddtags(tags))
+            self.__cache[rid]['metrics'][k] = v
+
+    def __tags_by_config(self, value):
+        # fmt: off
+        return {
+            'zid': value['zid'],
+            'name': value['metadata']['name'],
+            'zenoh_version': value['metadata']['zenoh_version']
+        }
+        # fmt: on
+
+    def __process_peers(self, value, tags):
+        peer_counts = defaultdict(lambda: 0)
+        for s in value:
+            peer_counts[s['whatami']] += 1
+
+        for k, v in peer_counts.items():
+            tags['whatami'] = k
+            self.count('sessions', v, tags=self.__map_to_ddtags(tags))
+
+    def check(self, instance):
+        url = instance.get('url')
+        try:
+            response = self.http.get(url + '/' + self.__PREFIX + '*?_stats=true')
+            response.raise_for_status()
+            data = response.json()
+
+            for item in data:
+                k = item['key'][len(self.__PREFIX) :]
+                rid = k[: self.__RID_LEN]
+                value = item['value']
+                gtags = self.__tags_by_config(value)
+                self.__process_stats(value['stats'], gtags.copy(), rid)
+                self.__process_peers(value['sessions'], gtags.copy())
+
+            self.service_check('can_connect', AgentCheck.OK)
+
+        except Timeout as e:
+            self.service_check(
+                'can_connect',
+                AgentCheck.CRITICAL,
+                message='Request timeout: {}, {}'.format(url, e),
+            )
+
+        except (HTTPError, InvalidURL, ConnectionError) as e:
+            self.service_check(
+                'can_connect',
+                AgentCheck.CRITICAL,
+                message='Request failed: {}, {}'.format(url, e),
+            )
+
+        except JSONDecodeError as e:
+            self.service_check(
+                'can_connect',
+                AgentCheck.CRITICAL,
+                message='JSON Parse failed: {}, {}'.format(url, e),
+            )
+
+        except BaseException as e:
+            self.service_check('can_connect', AgentCheck.CRITICAL, message=str(e))
```

Comment:
I'm not sure I see the value in handling these exceptions differently since in the end we're displaying the whole exception, and the exceptions we're catching include the URL.

So what our code is adding here is essentially a blurb before the exception that **kind of** summarizes it. Not blocking, but I'd simplify this to just one catch-all clause.